examples/plot_simulate_lfp.py

```diff
@@ -0,0 +1,130 @@
+params_fname = op.join(hnn_core_root, 'param', 'gamma_L5weak_L2weak.json')
+params = read_params(params_fname)
+params['tstop'] = 250
+###############################################################################
+# We will start with simulating the evoked response from
+# :ref:`evoked example <sphx_glr_auto_examples_plot_simulate_evoked.py>`.
+# We first instantiate the network.
+# (Note: Setting ``add_drives_from_params=True`` loads a set of predefined
+# drives without the drives API shown previously).
+net = Network(params)
+weights_ampa = {'L2_pyramidal': 0.0008, 'L5_pyramidal': 0.0075}
+synaptic_delays = {'L2_pyramidal': 0.1, 'L5_pyramidal': 1.0}
+rate_constant = {'L2_pyramidal': 140.0, 'L5_pyramidal': 40.0}
+net.add_poisson_drive(
+    'poisson', rate_constant=rate_constant, weights_ampa=weights_ampa,
+    location='proximal', synaptic_delays=synaptic_delays, seedcore=1079)
+
+###############################################################################
+# LFP recordings require specifying the electrode postion. It can be useful
+# to visualize the cells of the network to decide on the placement of each
+# electrode.
+net.plot_cells()
+
+###############################################################################
+# Electrode positions are stored under ``Network.lfp`` as a list
+# of tuples. Once we have chosen x,y,z coordinates for each electrode, we can
+# add them to the simulation.
+
+# XXX coordinates FUBAR, cortical depth direction is Y in NEURON objects!
+# electrode_pos = [(2, 400, 2), (6, 800, 6)]
+depths = list(range(-400, 1900, 200))
+electrode_pos = [(45, dep, 45) for dep in depths]
+net.add_electrode(electrode_pos)
+print(net.lfp)
+net.plot_cells()
+
+with MPIBackend(n_procs=4):
+    dpl = simulate_dipole(net)
+###############################################################################
+# Now that our electrodes are specified, we can run the simulation. The LFP
+# recordings are also stored under ``Network.lfp``.
+import matplotlib.pyplot as plt
+
+plt.figure()
+trial_idx = 0
+plt.plot(dpl[trial_idx].times, net.lfp[1]['data'][trial_idx])
+plt.plot(dpl[trial_idx].times, net.lfp[-2]['data'][trial_idx])
+plt.legend([f'e_pos {electrode_pos[1]}', f'e_pos {electrode_pos[-2]}'])
+plt.xlabel('Time (ms)')
+plt.ylabel(r'Potential ($\mu V$)')
+plt.show()
+
+from hnn_core.viz import _decimate_plot_data
+from hnn_core.dipole import _hammfilt
+from matplotlib.colors import SymLogNorm
+from mpl_toolkits.axes_grid1.inset_locator import inset_axes
+
+import numpy as np
+window_len = 10  # ms
+decimate = [5, 4]  # from 40k to 8k to 2k
+winsz = np.round(1e-3 * window_len * dpl[trial_idx].sfreq)
+
+plot_data = list()
+for elec in net.lfp:
+    filt_data = _hammfilt(elec['data'][trial_idx], winsz)
+    decim_data, decim_time, lfp_sfreq = _decimate_plot_data(
+        decimate, filt_data, dpl[trial_idx].times, sfreq=dpl[trial_idx].sfreq)
+    plot_data.append(decim_data)
+plot_data = np.array(plot_data)
```

Comment:
can `LFPElectrode` class take a list of tuples for electrode position? Then you can also have a function called `plot_csd`

Reply:
Proposing the `LFPArray` class instead